hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1023,15 +1028,21 @@ private long sleepWithBusyWait(long timeoutInMillis, long sleep) {
 
     @Override
     public boolean isMemberStateSafe() {
+        return getMemberState() == SAFE;
+    }
+
+    public InternalPartitionServiceState getMemberState() {
         if (hasOnGoingMigrationLocal()) {
-            return false;
+            return MIGRATION_LOCAL;
         }
+
         if (!node.isMaster()) {
             if (hasOnGoingMigrationMaster(Level.OFF)) {
-                return false;
+                return MIGRATION_ON_MASTER;
             }
         }
-        return isReplicaInSyncState();
+
+        return isReplicaInSyncState() ? SAFE : REPLICA_NOT_SYNC;
```

Comment:
There are 4 states. Node is making a migration by itself (MIGRATING_LOCAL), master is managing a migration on other nodes (MIGRATION_MASTER), there is a not-sync replica (REPLICA_NOT_SYNC), partition table is stable and works fine (SAFE). I couldn't come up with a better name than "SAFE" for the last case and actually I used it because the already existing method name is "isMemberStateSafe". I would like to hear your idea about it and happy to apply if you have a better name.